### DISCUSSION TITLE
My own custom-config keymaps for zmk-corne

### DISCUSSION THREAD

**Discussion by duanexavierbondad** [manna-harbour/miryoku#313] [Show and tell]
## Summary

It took me three days of experimentations and learning the in's and out's of ZMK and Miryoku but here we are now. First wireless build. Been using a custom made corne with QMK for about a year and a half now, trained to learn colemakdh usually an hour a day for 1 month until it is on my muscle memory. Now transitioning on Miryoku Layout to train on the symbols and numbers and layers. Throw in the transition to a wireless build with ZMK.

- Miryoku Implementations
  - [ ] Miryoku QMK
  - [X] Miryoku ZMK
  - [ ] Miryoku KMonad
  - [ ] Miryoku KMK
- Features
  - Layers
    - [X] Substitute Layer
    - [X] Add Layer
    - [ ] Remove Layer
    - [ ] Change Layer Display Name
  - Change Mapping
    - [X] All Layers
    - [ ] Per Layer
   - Changes
     - Updated ColemakDH to have the semicolon ( ; ), Put _single qoute [ ' ]_ on the **Number Layer** and _double quotes [ " ]_ on the **Symbol Layer.** 
     - Swap positions of Tab and Space | Swap positions of Backspace and Enter. Applied to BASE, TAP, and BASE_QWERTY layers
     - Added a Valorant Gaming Layer for Left side.
     
    - Future Experimentations
      - **enable mouse emulation** since there are still errors on my build when adding the #define MIRYOKU_KLUDGE_MOUSEKEYSPR. It is unable to fine the #include <behaviors/mouse_move.dtsi> | No such file or directory. Will update this once I am able to rebuild.
      - **LED underglows**: It seems I am unable to run underglows with miryoku even in the QMK firmware when I tried to flash it. 

    [custom_config.h](https://github.com/duanexavierbondad/miryoku_zmk_xave/blob/master/miryoku/custom_config.h)

**Comment by manna-harbour** [comment]
Thanks for sharing your config!

> Now transitioning on Miryoku Layout to train on the symbols and numbers and layers. 

Consider drilling numbers and symbols on monkeytype.

> enable mouse emulation since there are still errors on my build when adding the #define MIRYOKU_KLUDGE_MOUSEKEYSPR. It is unable to fine the #include <behaviors/mouse_move.dtsi> | No such file or directory. Will update this once I am able to rebuild.

You also need to specify a branch containing the mousekeys PR. See https://github.com/manna-harbour/miryoku_zmk#mousekeys-pr.

>LED underglows: It seems I am unable to run underglows with miryoku even in the QMK firmware when I tried to flash it.

For ZMK and QMK you'll need to enable RGB in the build configuration. 

For ZMK see https://github.com/manna-harbour/miryoku_zmk#kconfig-configuration. For QMK see the Custom Hardware Support section at https://github.com/manna-harbour/miryoku/discussions/85.
